Otus-DevOps-2020-02/DmitryKorlas_microservices
# Logging 1

# Completed the homework №23


## What were done:
 during this homework, the external logging and it visualisation has been added to the service. We used **elasticesarch** + **fluentd** + **kibana** for it. Moreover, **zipkin** tool has been added for distributed request tracing.

**done:**
 - update app files for logging homework from https://github.com/express42/reddit/tree/logging
 - add fluentd container to aggregate+parse logs
 - use fluentd and kibana as external logs storage
 - add filter into fluentd config to see more detailed info about log entry in kibana
 - add regexp-based and grok-based log parsers for unstructured logs
 - add zipkin service for requests tracing
 - (the task *): find out why the post page works too slow
   > method `find_post` contains `time.sleep(3)` - this is why it's processed too long

## How to run the project:
  ```sh
  cd docker
  cp .env.example env
  docker-compose -f docker-compose-logging.yml -f docker-compose.yml up -d
  ```

## How to check, that it works correctly:
 - after start the app service, navigate to homepage to produce requests which should be logged.
 - see logs in Kibana, navigate to http://<DOCKER_MACHINE_IP>:5601
 - see zipkin control panel at http://<DOCKER_MACHINE_IP>:9411

## PR checklist:
 - [x] Added a label with the number of homework
